### PR TITLE
adjust sel of frozen models

### PR DESCRIPTION
**Pull request by njzjz**
This PR adds the folling features:

- save checkpoints when `numb_steps == 0`
- fix a typo in #1482
- restore `sel` in the graph
- support recovering when sel != original_sel (only when `set_davg_zero` is `true`)

So, when `set_davg_zero` is `true`, one can use `dp train input.json -f graph.pb` to get a new model, where `input.json` decreased or increased `sel` and set `numb_steps` to zero, and then freeze the model. Two models should get the same result (descriptors, energies, forces) when sel is enough for both models. The UT confirms it.
As discussed, the theory behind it is also clear.

Here we may have the following motivations to use this feature:
(1) One can train a model that sel fits the training data to get the best performance during training, and then increase the sel, to prevent sel is not enough;
(2) One can continue to train a small-sel model against the large-sel data, without retraining parameters;
(3) One can train a model for mixed systems with different sel, but run simulations for each system with small-sel model (even 0 if a system does not contain a element) to get best performance.

### CODE REVIEW

**Review by wanghan-iapcm — APPROVED** [review]
LGTM

**Comment by wanghan-iapcm** [comment]
Could you add the introduction on how to use this feature into doc (e.g. doc/train/training-advanced.md)?

**Comment by njzjz** [comment]
Sure.

**Review by wanghan-iapcm — APPROVED** [review]

File: deepmd/descriptor/se_a.py [hunk 1]
```diff
@@ -827,7 +830,12 @@ def _filter(
           # inputs_reshape = tf.reshape(inputs, [-1, shape[1]//4, 4])
           # natom x 4 x outputs_size
           # xyz_scatter_1 = tf.matmul(inputs_reshape, xyz_scatter, transpose_a = True)
-          xyz_scatter_1 = xyz_scatter_1 * (4.0 / shape[1])
+          if self.original_sel is None:
+              # shape[1] = nnei * 4
+              nnei = shape[1] / 4
+          else:
+              nnei = np.sum(self.original_sel)
```
Comment (njzjz) [inline deepmd/descriptor/se_a.py:837]: I realized that adjusting `sel` again would lose this information.
We need to restore `nnei` forever.

**Pull request merged**